frontend/src/api/network/notebooks.ts

```diff
@@ -256,12 +262,19 @@ export const updateNotebook = (
   data.notebookId = existingNotebook.metadata.name;
   const notebook = assembleNotebook(data, username);
 
+  const oldNotebook = { ...existingNotebook };
```

Comment:
You'll need to use `_.cloneDeep` unfortunately... spreads are shallow copies and thus when you dig into properties like `.spec` that's an object reference -- so you are now mutating the previous object.